src/prog_translator.cc

```diff
@@ -55,9 +57,22 @@ void ProgTranslator::Init() {
   ctx_.m = CoinPackedMatrix(false, 0, 0);
 }
 
+void ProgTranslator::CheckPref(double pref) {
+  if (pref <= 0) {
```

Comment:
I can add to the error message. This is also holdover from days before deprecating 0-valued preferences.